CIP-0057/README.md

```diff
@@ -0,0 +1,414 @@
+---
+CIP: 57
+Title: Plutus Contract Blueprint
+Authors: KtorZ <matthias.benkort@cardanofoundation.org>, scarmuega <santiago@carmuega.me>
+Discussions-To: https://github.com/cardano-foundation/CIPs/pull/258
+Status: Draft 
+Type: Process
+Created: 2022-05-15
+License: CC-BY-4.0
+---
+
+## Abstract
+
+This document specifies a language for documenting Plutus contracts in a machine-readable manner. This is akin to what [OpenAPI](https://swagger.io/specification) or [AsyncAPI](https://www.asyncapi.com/docs/specifications/v2.4.0) are for, documenting HTTP services and asynchronous services respectively. In a similar fashion, A Plutus contract has a binary interface which is mostly defined by its datum and redeemer. Since all interactions with contracts are done through transactions, we can also model _endpoints_ of a contract as transitions from a known state to another purely declaratively. 
+
+This document is therefore a meta-specification defining the vocabulary and validation rules with which one can specify a Plutus contract interface, a.k.a **Plutus contract blueprint**.
```

Comment:
To make sure I understand the wording here, you want to define a language in which you can define the behaviour of your protocol?

Reply:
Not the behavior (though to some extend, this is a stretch goal / bonus), but the application binary interface. That is, what other programs need to know in order to interact with a particular Plutus contract.